### DISCUSSION TITLE
Do vscode-go now have nothing to do with dlv-dap?

### DISCUSSION THREAD

**Discussion by xujintao** [golang/vscode-go#1817] [Discussions]
Accourding to [Go Debug Extension Architecture Overview](https://github.com/golang/vscode-go/blob/master/docs/debugging.md#go-debug-extension-architecture-overview), since legacy debug adapter has been removed from vscode-go and the vscode's Generic Debugger UI can communicate with Delve(dlv-dap) using DAP directly, does vscode-go now have nothing to do with dlv-dap?

**Comment by hyangah** [comment]
The legacy debug adapter is still in use for remote debugging (attach+remote mode) and users who need to use the legacy debug adapter to overcome its limitations (e.g. requires delve 1.6.1+) We will eventually get rid of `dlv-dap` but use `dlv`, but not right now yet.

**Comment by firelizzard18** [comment]
Integrating a debug adapter with VSCode still requires an extension. In the simplest case, all the extension does is define debug configuration and pass the binary path and args to VSCode, but the extension is still required.

**Reply by xujintao** [reply]
Great, thanks.
